x/blob/test/decode_blob_tx_test.go

```diff
@@ -0,0 +1,111 @@
+package test
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/pkg/blob"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type BlockResponse struct {
+	Block Block `json:"block"`
+}
+
+type Block struct {
+	Data BlockData `json:"data"`
+}
+
+type BlockData struct {
+	Txs []string `json:"txs"`
+}
+
+// TestDecodeBlobTx demonstrates how one can take the response from the Comet
+// BFT API endpoint /cosmos/base/tendermint/v1beta1/blocks/{block_number} and
+// convert a base64 encoded transaction into an sdk.Tx.
+//
+// NOTE: this process differs from other Cosmos SDK chains because the
+// transactions of type BlobTx won't be usable directly. One needs to extract
+// the Tx field inside the BlobTx prior to decoding it as an sdk.Tx.
+func TestDecodeBlobTx(t *testing.T) {
+	blockResponse := getBlockResponse(t)
+
+	for i, rawTx := range blockResponse.Block.Data.Txs {
+		decoded, err := base64Decode(rawTx)
+		if err != nil {
+			t.Errorf("error decoding base64 tx: %v", err)
+		}
+
+		txBytes := getTxBytes(decoded)
+		tx, err := decodeTx(txBytes)
+		if err != nil {
+			t.Errorf("error decoding tx: %v", err)
+		}
+
+		if i == 273 {
+			// https://celenium.io/tx/C55BDD3DF3348A9F8D9206528051804754F009A1B9D0F69CCC2F9D4334215D21
+			wantHash := "C55BDD3DF3348A9F8D9206528051804754F009A1B9D0F69CCC2F9D4334215D21"
+			gotHash := strings.ToUpper(hex.EncodeToString(sum(txBytes)))
+			assert.Equal(t, wantHash, gotHash)
+
+			wantSigner := "celestia18y3ydyn7uslhuxu4lcm2x83gkdhrrcyaqvg6gk"
+			gotSigner := tx.GetMsgs()[0].GetSigners()[0].String()
+			assert.Equal(t, gotSigner, wantSigner)
+		}
+	}
+}
+
+func getBlockResponse(t *testing.T) BlockResponse {
```

Comment:
A func description is helpful as the name of the function sounds like it retrieves live block data.

Reply:
4942be3e963d0936043f31d5161ae6aceda8a94b